signatories.sympl

```diff
@@ -1,29 +1,62 @@
-schema Contract:
-    id: str
-    text: str
-    creator: KeyAlias
-    channel_id: ChannelName
-    parties: List[KeyAlias]
-    signatures: List[Signature]
+# Signature is applied to a Contract by one of the Contract's parties
 
 schema Signature:
-    id: str
-    contract_id: str
+    id: Identifier
+    contract_id: Identifier
     party: KeyAlias
     created_at: DateTime
+# Contract is an agreement that can be entered into by parties.
+
+schema Contract:
+    id: str
+    text: str
+    creator: str
+    parties: Optional[List[KeyAlias]]
+    signatures: Optional[List[Signature]]
+# ContractChannel is stored on PUBLIC and is where ChannelName where a Contract is
+# stored are looked up
+
+schema ContractChannel:
+    contract_id: Identifier
+    channel: ChannelName
+CONTRACT_PREFIX : str = "CONTRACT"
 
 @clientside
-def contract() -> Contract:
-    return Contract
+def contract() -> None:
+    return None
 
 @clientside
-def contract_create() -> Contract:
-    return Contract
+def contract_create(text: str) -> None:
+    cvm.job_start()
+    ch : ChannelName = cvm.new_channel(CONTRACT_PREFIX)
+    with PostTxArgs(ch):
+        _contract_create(text)
```

Comment:
I need some help with this concept.

I want each `Contract` to be stored in its own private channel. So, I create a new channel and call `PostTxArgs(ch)`

Then I call the @executable where I get the `ChannelName` with `cvm.tx.write.channel`

But in the @executable, I need to change contexts. After writing the Contract to its own private channel, I need to write the ContractChannel (basically a hashmap) to PUBLIC.

How do I change contexts so that the subsequent cvm.storage.put is going to PUBLIC storage?

Reply:
(I don't know why md capitalized the X in executable)